### DISCUSSION TITLE
How to disable use of user and global config?

### DISCUSSION THREAD

**Discussion by ssbarnea** [gitpython-developers/GitPython#1212] [Q&A]
I posted the same question on SO https://stackoverflow.com/questions/67109964/how-to-force-gitpython-to-ignore-global-git-configuration but i think it would make sense to repeat it here, being specific to the project.

While I was able to [find an ugly workaround for forcing git to ignore its system or user configuration](https://stackoverflow.com/questions/43881807/how-to-tell-git-to-ignore-global-config), I was not able to achieve the same while using gitpython.

```
GIT_CONFIG_NOSYSTEM=1 XDG_CONFIG_HOME=/ HOME=/ git ...
```

How can I achieve the same using gitpython library?

Note that I tried hacking os.environ but it appears to not work. If anyone wonders why I would want this is just because I want to have a portable `is_dirty()` implementation, one that does not consider user ignores as I want developer to add required ignores if those are needed.

**Comment by Byron** [comment]
More information would be required to answer this as it depends on how you are using GitPython.

In short, if `repo.git.command(…)` syntax is used that can be affected with environment variables and behave accordingly.

If the built in method is used…

https://github.com/gitpython-developers/GitPython/blob/651a81ded00eb993977bcdc6d65f157c751edb02/git/repo/base.py#L667

…a PR might be required to make it respect environment variables. It might already be possible to affect it by using this context manager though.

https://github.com/gitpython-developers/GitPython/blob/651a81ded00eb993977bcdc6d65f157c751edb02/git/cmd.py#L872
